src/pages/preferences/preferencesTabs/publicKeysTab/EditModal.tsx

```diff
@@ -102,7 +102,10 @@ export const EditModal: React.VFC<EditModalProps> = ({
     }
     const hasRSA = keyValue?.substring(0, SSH_RSA.length) === SSH_RSA;
     const hasDSS = keyValue?.substring(0, SSH_DSS.length) === SSH_DSS;
-    if (!hasRSA && !hasDSS) {
+    const hasED25519 =
+      keyValue?.substring(0, SSH_ED25519.length) === SSH_ED25519;
+    const hasECDSA = keyValue?.substring(0, SSH_ECDSA.length) === SSH_ECDSA;
+    if (!hasRSA && !hasDSS && !hasED25519 && !hasECDSA) {
```

Comment:
Can we just combine these into a regexp? 
Instead of iterating through each type of key can we just have one check that validates the ssh key.
```ts
const isValidKey = keyValue.match(validSSHKey).length > 0;

const validSSHKey = /someSSHKeyRegex/

```